### PR TITLE
Slightly better perf by removing some duplicate casts

### PR DESCRIPTION
**Pull request by SimonCropp**

### CODE REVIEW

**Review by maxkatz6 — APPROVED** [review]

**Comment by maxkatz6** [comment]
Thanks!
But it will take a while to merge all of these PRs, as github+azure integration can't chain merge requests.

**Comment by SimonCropp** [comment]
@maxkatz6 i will back off a bit

File: src/Avalonia.Base/Controls/NameScopeExtensions.cs [hunk 1]
```diff
@@ -25,13 +25,18 @@ public static class NameScopeExtensions
 
             var result = nameScope.Find(name);
 
-            if (result != null && !(result is T))
+            if (result == null)
             {
-                throw new InvalidOperationException(
-                    $"Expected control '{name}' to be '{typeof(T)} but it was '{result.GetType()}'.");
+                return (T?)result;
```
Comment (maxkatz6) [inline src/Avalonia.Base/Controls/NameScopeExtensions.cs:30]: Makes sense to return `null` instead.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0028262-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

**Pull request merged**